pkg/controllermanager/controller/shoot/shoot_care_control.go

```diff
@@ -218,7 +218,7 @@ func garbageCollection(initShootClients func() error, botanist *botanistpkg.Bota
 		}
 	}()
 
-	if !botanist.Shoot.IsHibernated {
+	if !botanist.Shoot.HibernationEnabled {
```

Comment:
The health check should only be executed if `.status.hibernated=false`. Can you please do this?